front-end/src/components/main-page/OverrideModal.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Modal from 'react-modal';
+import BaseButton from '../general/BaseButton';
+import PropTypes from 'prop-types';
+
+class OverrideModal extends React.Component {
+
+    static propTypes = {
+        show: PropTypes.bool.isRequired,
+        toggleModal: PropTypes.func.isRequired,
+        uploadFileWithOverride: PropTypes.func.isRequired,
+    }
+
+    async overrideFile(){
+        this.props.uploadFileWithOverride();
+        this.props.toggleModal();
+    }
+
+    componentDidMount() {
+        Modal.setAppElement("body");
+    }
+
+    render() {
+        return (
+            <Modal
+                isOpen={this.props.show}
+                className="container w-40 md:w-60 mx-auto bg-purple-100 rounded-lg shadow-lg py-3 mt-[40vh] flex flex-col"
+                onRequestClose={this.props.toggleModal}
+                shouldCloseOnEsc={true}
+                ariaHideApp={false}
+            >
+                <h2 className="justify-center flex m-3" data-testid="override-modal">
```

Comment:
Even though the tailwind styling has justify-center, in the modal, it doesn't look perfectly centered (little nitpicks, but they're noticeable enough) so maybe fiddle around just a little with the margins and padding and stuff to make things look nice and neat?